### PR TITLE
fix: [#6825] Update xunit and fix compatibility issues in unit tests

### PR DESCRIPTION
**Pull request by JhontSouth**
Addresses # 6825
#minor

## Description
This PR updates _**xunit**_ package version from 2.4.1 to 2.9.0 and applies the respective changes to meet the new version requirements. 

## Specific Changes
  - Changed _async void_ methods to _async Task_.
  - Swapped assert equal parameters to leave the expected value in the correct argument position. 
  - Changed assert equal comparison of size collection 0 and 1 to _Empty_ and _Single_ methods.
  - Added async operators instead of the use of Wait, GetAwaiter, and Result methods.

## Testing
The following image shows the unit tests working in Net 6.0 and 8.0.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/EntityRecognizerTests.cs [hunk 1]
```diff
@@ -41,40 +42,61 @@ public class EntityRecognizerTests
...
+            var results = await recognizers.Value.RecognizeEntitiesAsync(turnContext);
 
             Assert.Equal(6, results.Count);
             Assert.Single(results.Where(entity => entity.Type == "age").ToList());
         }
 
         [Fact]
+
+/* Unmerged change from project 'Microsoft.Bot.Builder.Dialogs.Adaptive.Tests (net8.0)'
+Before:
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/EntityRecognizerTests.cs:57]: Same in this file

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs [hunk 2]
```diff
@@ -65,13 +65,13 @@ public async void SendActivitiesAsyncShouldSucceedAndNoActivityReturnedWithActiv
             twilioApi.Setup(x => x.SendMessageAsync(It.IsAny<TwilioMessageOptions>(), default)).Returns(Task.FromResult(resourceIdentifier));
 
             var twilioAdapter = new TwilioAdapter(twilioApi.Object, _adapterOptions);
-            var resourceResponses = await twilioAdapter.SendActivitiesAsync(null, new Activity[] { activity.Object }, default).ConfigureAwait(false);
+            var resourceResponses = await twilioAdapter.SendActivitiesAsync(null, new Activity[] { activity.Object }, default);
```
Comment (sw-joelmut) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs:68]: I hope removing the `.ConfigureAwait(false)` doesnt complain again at some point, maybe in the tests were not necessary? but in the libraries it always complain, i dont know if there is some rule.
Although, it looks better this way, less code.

File: tests/Microsoft.Bot.Connector.Tests/Authentication/ManagedIdentityAuthenticatorTests.cs [hunk 3]
```diff
@@ -20,7 +20,14 @@ public class ManagedIdentityAuthenticatorTests
         private readonly Func<string, string> audience = (id) => $"audience {id} ";
 
         [Fact]
+
+/* Unmerged change from project 'Microsoft.Bot.Connector.Tests (net6.0)'
+Before:
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Connector.Tests/Authentication/ManagedIdentityAuthenticatorTests.cs:25]: same here

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Connector.Tests/Authentication/ManagedIdentityAuthenticatorTests.cs (hunk 3 above)
Comment (JhontSouth) [reply tests/Microsoft.Bot.Connector.Tests/Authentication/ManagedIdentityAuthenticatorTests.cs]: fixed, I removed all the auto generated comments

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs (hunk 2 above)
Comment (JhontSouth) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs:68]: yes, following the doc of xnuit, this is not needed in the unit tests

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /6836

**Pull request closed without merging**